### PR TITLE
bug fix for update management dynamic group

### PR DESCRIPTION
**Pull request by jemex**
## Description

bug fix for creating Software update configuration with dynamic groups

## Checklist

- [x ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x ] The title of the PR is clear and informative
- [x ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.ps1 [hunk 1]
```diff
@@ -196,4 +196,266 @@ $query1Scope = @(
...
+    $s = New-AzAutomationSchedule -ResourceGroupName $rg `
+                                       -AutomationAccountName $aa `
+                                       -Name $name `
+                                       -Description test-OneTime `
+                                       -OneTime `
+                                       -StartTime $startTime `
+                                       -ForUpdate
+
+$query1Scope = @(
+        "/subscriptions/cd45f23b-b832-4fa4-a434-1bf7e6f14a5a/resourceGroups/mms-wcus"
```
Comment (cormacpayne) [inline src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.ps1]: @jemex as mentioned in the previous PR, we should avoid using hard-coded resource ids where we can -- if it's determined that there is specific setup needed for the test, a comment should be left before the test to outline what the user needs to do in order to properly setup their environment to re-record this test

File: src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs [hunk 2]
```diff
@@ -39,9 +39,12 @@ public SoftwareUpdateConfiguration CreateSoftwareUpdateConfiguration(string reso
                     {
 
                         var tags = new Dictionary<string, IList<string>>();
-                        foreach (var tag in query.TagSettings.Tags)
+                        if (query.TagSettings != null && query.TagSettings.Tags != null && query.TagSettings.Tags.Count > 0)
```
Comment (cormacpayne) [inline src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs]: @jemex nit: the last check (`query.TagSettings.Tags.Count > 0`) shouldn't be necessary here

File: src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateConfiguration.cs [hunk 3]
```diff
@@ -100,19 +100,21 @@ internal SoftwareUpdateConfiguration(string resourceGroupName, string automation
                 foreach (var query in suc.UpdateConfiguration.Targets.AzureQueries)
                 {
                     var tags = new Dictionary<string, List<string>>();
-                    foreach (var tag in query.TagSettings.Tags)
+                    if(query.TagSettings.Tags != null)
```
Comment (cormacpayne) [inline src/Automation/Automation/Model/UpdateManagement/SoftwareUpdateConfiguration.cs]: @jemex do we want to also include a null check on `query` and `query.TagSettings` here?

File: src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.ps1 (hunk 1 above)
Comment (jemex) [reply src/Automation/Automation.Test/ScenarioTests/UpdateDynamicGroupPrePostTests.ps1]: @cormacpayne  I  have added a pre-req steps for re-recording all the tests on the top of the test file

**Pull request merged**